File: src/src/main/java/com/mewebstudio/translatable/dto/response/CategoryTranslationResponse.java
package com.mewebstudio.translatable.dto.response;

import com.mewebstudio.translatable.entity.CategoryTranslation;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CategoryTranslationResponse {
    private String name;
    private String description;
    private CategoryResponse category;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public static CategoryTranslationResponse convert(CategoryTranslation translation, boolean includeOwner) {
        return CategoryTranslationResponse.builder()
            .name(translation.getName())
            .description(translation.getDescription())
            .category(includeOwner ? CategoryResponse.convert(translation.getOwner()) : null)
            .createdAt(translation.getCreatedAt())
            .updatedAt(translation.getUpdatedAt())
            .build();
    }
}

File: src/main/kotlin/com/mewebstudio/translatable/dto/response/CategoryTranslationResponse.kt
package com.mewebstudio.translatable.dto.response

import com.mewebstudio.translatable.entity.CategoryTranslation
import java.time.LocalDateTime

data class CategoryTranslationResponse(
    val name: String,

    val description: String?,

    val category: CategoryResponse?,

    val createdAt: LocalDateTime,

    val updatedAt: LocalDateTime
) {
    companion object {
        fun convert(translation: CategoryTranslation, includeOwner: Boolean = false): CategoryTranslationResponse =
            CategoryTranslationResponse(
                name = translation.name,
                description = translation.description,
                category = if (includeOwner) {
                    CategoryResponse.convert(translation.owner)
                } else {
                    null
                },
                createdAt = translation.createdAt,
                updatedAt = translation.updatedAt
            )
    }
}
